Client/Frontend/Browser/New Tab Page/NewTabPageViewController.swift

```diff
@@ -424,8 +455,167 @@ class NewTabPageViewController: UIViewController, Themeable {
         notificationController = nil
     }
     
+    // MARK: - Brave Today
+    
+    private func handleBraveTodayAction(_ action: BraveTodaySectionProvider.Action) {
+        switch action {
+        case .welcomeCardAction(.closedButtonTapped):
+            Preferences.BraveToday.isShowingIntroCard.value = false
+            if let section = layout.braveTodaySection, collectionView.numberOfItems(inSection: section) != 0 {
+                collectionView.deleteItems(at: [IndexPath(item: 0, section: section)])
+            }
+        case .welcomeCardAction(.learnMoreButtonTapped):
+            delegate?.navigateToInput("https://brave.com/privacy/#brave-today", inNewTab: false, switchingToPrivateMode: false)
+        case .welcomeCardAction(.settingsButtonTapped),
+             .emptyCardTappedSourcesAndSettings:
+            tappedBraveTodaySettings()
+        case .errorCardTappedRefresh:
+            loadFeedContents()
+        case .itemAction(.opened(let inNewTab, let switchingToPrivateMode), let context):
+            guard let url = context.item.content.url else { return }
+            delegate?.navigateToInput(
+                url.absoluteString,
+                inNewTab: inNewTab,
+                switchingToPrivateMode: switchingToPrivateMode
+            )
+        case .itemAction(.toggleSource, let context):
+            let isEnabled = feedDataSource.isSourceEnabled(context.item.source)
+            feedDataSource.toggleSource(context.item.source, enabled: !isEnabled)
+            collectionView.reloadData()
```

Comment:
Do we have to reload data here? We do not hide or show the item once it's toggled, or is it future-proof code

Reply:
Might be old code, I will double check if its still needed